xjc-kafka-connect-plugin/src/main/java/com/github/jcustenborder/kafka/connect/xml/KafkaConnectPlugin.java

```diff
@@ -154,7 +172,7 @@ Map<JType, StaticTypeState> buildTypeLookup(JCodeModel codeModel) {
     add(result, this.types.schemaBuilder().staticInvoke("int64"), this.types.schema().staticRef("INT64_SCHEMA"), "toInt64", "fromInt64BigInteger", codeModel, BigInteger.class);
     add(result, this.types.schemaBuilder().staticInvoke("bytes"), this.types.schema().staticRef("BYTES_SCHEMA"), "toBytes", "fromBytes", codeModel, byte[].class);
     add(result, this.types.schemaBuilder().staticInvoke("string"), this.types.schema().staticRef("STRING_SCHEMA"), "toString", "fromString", codeModel, String.class);
-    add(result, this.types.decimal().staticInvoke("builder").arg(JExpr.lit(12)), null, "toDecimal", "fromDecimal", codeModel, BigDecimal.class);
+    add(result, this.types.decimal().staticInvoke("builder").arg(JExpr.lit(this.decimalScale)), null, "toDecimal", "fromDecimal", codeModel, BigDecimal.class);
```

Comment:
to preserve backward compatibility the default value of this.decimalScale should be set somewhere to the old fixed value of 12